patroni/ctl.py

```diff
@@ -1372,3 +1373,205 @@ def format_pg_version(version):
         return "{0}.{1}.{2}".format(version // 10000, version // 100 % 100, version % 100)
     else:
         return "{0}.{1}".format(version // 10000, version % 100)
+
+
+def enrich_config_from_running_instance(dsn: str, config: Dict[str, Any], no_value_msg: str) -> None:
+    from patroni.postgresql.config import parse_dsn
+    from patroni.config import _AUTH_ALLOWED_PARAMETERS
+
+    parsed_dsn = parse_dsn(dsn)
+    if not parsed_dsn:
+        raise PatroniCtlException('Failed to parse DSN string')
+    if not parsed_dsn.get('user'):
+        raise PatroniCtlException('No user provided in the DSN')
+
+    from . import psycopg
+    conn = psycopg.connect(dsn=dsn)
+
+    with conn.cursor() as cur:
+        cur.execute("SELECT 1 FROM pg_roles WHERE rolname=%s AND rolsuper='t';", (parsed_dsn['user'],))
+        if cur.rowcount < 1:
+            conn.close()
+            raise PatroniCtlException('The provided user does not have superuser privilege')
+
+        cur.execute("SELECT name, setting \
+                     from pg_settings \
+                     where setting <> reset_val and setting <> '(disabled)' \
+                     or name in ('port', 'listen_addresses');")
+
+        # adjust values
+        config['postgresql']['parameters'] = dict()
+        for p, v in cur.fetchall():
+            if p in config['bootstrap']['dcs']['postgresql']['parameters']:
+                config['bootstrap']['dcs']['postgresql']['parameters'][p] = v
+            else:
+                config['postgresql']['parameters'][p] = v
+
+        # gather additional info
+        cur.execute("SELECT setting from pg_settings where name = 'data_directory';")
+        config['postgresql']['data_dir'] = cur.fetchone()[0]
+
+        cur.execute("SELECT pg_read_file((SELECT setting FROM pg_settings WHERE name='hba_file'));")
+        pg_hba = cur.fetchone()[0]
+        config['postgresql']['pg_hba'] = [i for i in pg_hba.split(chr(10))
+                                          if i.startswith(('local',
+                                                           'host',
+                                                           'hostssl',
+                                                           'hostnossl',
+                                                           'hostgssenc',
+                                                           'hostnogssenc'))]
+
+    conn.close()
+
+    if not config['bootstrap']['dcs']['postgresql']['bin_dir']:
```

Comment:
It is `postgresql.bin_dir`.

Reply:
🤦‍♀️  thanks

Reply:
hm. but why is it here then?

https://github.com/zalando/patroni/blob/master/patroni/config.py#L183-L201

also this reminded me to check if watchdog is actually configurable via dynamic config

Reply:
Oversight of https://github.com/zalando/patroni/pull/263/files#diff-fc085851aad448e6907a4dcc93e39aa0c3aff122e1d11ccf955252ba2019ccbbR45 :(